src/extension/feature-pregeneralized/src/main/java/org/geoserver/data/gen/info/GeneralizationInfosProviderImpl.java

```diff
@@ -50,4 +57,39 @@ protected URL deriveURLFromSourceObject(Object source) throws IOException {
         }
         throw new IOException("Cannot read from " + source);
     }
+
+    @Override
```

Comment:
Please add a javadoc explaining why it is required to override this method. Ideally also on `deriveURLFromSourceObject()`.